src/app/auth/authentication.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { Router } from "@angular/router";
+import { Logger } from '../shared/logger.service';
+
+@Injectable()
+export class AuthenticationService {
+  private authToken: any = null;
+
+  constructor(private router: Router, private logger: Logger) {
+    //If a token is passed part of the url - extract that and set it    
+    if((window.location.href).indexOf('?token=')>0){
```

Comment:
What if `token` won't come as first query param? Wouldn't it be better to extract all first into some sort of map and get relevant by key?